src/main/java/build/buildfarm/worker/ExecuteActionStage.java

```diff
@@ -29,6 +29,7 @@
   private final Set<Thread> executors = Sets.newHashSet();
   private final AtomicInteger executorClaims = new AtomicInteger(0);
   private BlockingQueue<OperationContext> queue = new ArrayBlockingQueue<>(1);
+  private int size = 0;
```

Comment:
should probably be volatile, accessed from multiple threads without regard for sync